src/core/core.scale.js

```diff
@@ -569,6 +569,60 @@ module.exports = function(Chart) {
 				0;
 		},
 
+		/**
+		 * Returns a subset of ticks to be plotted.
+		 * This functionality makes it so that we don't have overlapping labels.
+		 */
+		autoSkip: function() {
+			var skipRatio;
+			var me = this;
+			var isHorizontal = me.isHorizontal();
+			var optionTicks = me.options.ticks.minor;
+			var ticks = me.getTicks();
+			var tickCount = ticks.length;
+			var labelRotationRadians = helpers.toRadians(me.labelRotation);
+			var cosRotation = Math.cos(labelRotationRadians);
+			var longestRotatedLabel = me.longestLabelWidth * cosRotation;
+			var result = [];
+
+			// figure out the maximum number of gridlines to show
+			var maxTicks;
+			if (optionTicks.maxTicksLimit) {
+				maxTicks = optionTicks.maxTicksLimit;
+			}
+
+			if (isHorizontal) {
+				skipRatio = false;
+
+				if ((longestRotatedLabel + optionTicks.autoSkipPadding) * tickCount > (me.width - (me.paddingLeft + me.paddingRight))) {
+					skipRatio = 1 + Math.floor(((longestRotatedLabel + optionTicks.autoSkipPadding) * tickCount) / (me.width - (me.paddingLeft + me.paddingRight)));
+				}
+
+				// if they defined a max number of optionTicks,
+				// increase skipRatio until that number is met
+				if (maxTicks && tickCount > maxTicks) {
+					while (!skipRatio || tickCount / (skipRatio || 1) > maxTicks) {
```

Comment:
We don't know how long it will take to get it fully rewritten, neither if it will land in 2.7.0. Let's take time to cleanup the code while refactoring it. Else there is no need for this kind of PR and I would prefer to merge the final one with a full overview of the new implementation.